corehq/apps/app_manager/static/app_manager/js/details/case_claim.js

```diff
@@ -352,16 +356,36 @@ hqDefine("app_manager/js/details/case_claim", function () {
                 hidden: searchProperty.hidden,
                 receiverExpression: searchProperty.receiver_expression,
                 itemsetOptions: searchProperty.itemset,
+                isGroup: searchProperty.is_group,
+                groupKey: searchProperty.group_key,
             }, saveButton);
         });
 
         self.search_properties = ko.observableArray(
             wrappedSearchProperties.length > 0 ? wrappedSearchProperties : [searchPropertyModel({}, saveButton)]
         );
 
+        self.search_properties.subscribe(function (newProperties) {
+            let groupKey = '';
+            ko.utils.arrayForEach(newProperties, function (property, index) {
+                if (property.isGroup) {
+                    groupKey = `group_header_${index}`;
+                    if (property.name !== groupKey) {
+                        property.name(groupKey);
+                    }
+                }
+                if (property.groupKey !== groupKey) {
+                    property.groupKey = groupKey;
```

Comment:
This seems like it will reset `property.groupKey` to an empty string for every property that's not a group itself (i.e. where `property.isGroup` is false), right? Unless the `isGroup` property is known to always come first and can set ``groupKey = `group_header_${index}`;``

Reply:
Ya that's the gist. The `newProperties` array will be in the same order as what's displayed on the UI side (top to bottom). So this sets every non-group property's `groupKey` to equal the most recent group property's groupKey. 

But I don't love the way this reads either. The checks `if (property.name !== groupKey)` and `if (property.groupKey !== groupKey) ` are just to avoid updating the item if there's no change. Otherwise, it'll trigger the `isEnabled` computed observable everytime the item is updated.